pace-util/pace/util/initialization/allocator.py

```diff
@@ -80,3 +80,12 @@ def _allocate(
             origin=origin,
             extent=extent,
         )
+
+    def get_origin(self, dims: Iterable[str]) -> Tuple[int, ...]:
+        return self._sizer.get_origin(dims=dims)
+
+    def get_extent(self, dims: Iterable[str]) -> Tuple[int, ...]:
+        return self._sizer.get_extent(dims=dims)
+
+    def get_shape(self, dims: Iterable[str]) -> Tuple[int, ...]:
+        return self._sizer.get_shape(dims=dims)
```

Comment:
I didn't propagate this change to the code accessing `._sizer`, I'll fix that.

Because of the [Law of Demeter](https://en.wikipedia.org/wiki/Law_of_Demeter), which I didn't know when writing the first version of this code. There are several parts of pace-util where I broke this guideline. Fortunately updating the code to follow this guideline can always be done in a way that is backwards-compatible, like here.

The benefit basically is if you know the dependent classes of a bit of code, you only need to know the methods on those classes or data contained on those classes to know everything the code might be doing. In this version, it's possible to write a QuantityFactory that doesn't use a SubtileGridSizer, whereas if the `.sizer` attribute is public on QuantityFactory then we're saying every QuantityFactory has to provide a SubtileGridSizer. That can make it a bit harder to make a polymorphic substitute of the class, e.g. if someone wanted to write an alternate QuantityFactory that makes tensorflow-backed Quantities.

There's a second reason here, which is we shouldn't access private attributes from outside a class. We were manually accessing `._sizer`. Instead of making sizer public, I added methods, since that's what the Law of Demeter would suggest.

I probably should have done this in another PR, but fortunately the changes are pretty small (other than here, there are 4 code lines I'll be updating shortly).

Reply:
Updated. Turns out we do still need `.sizer` to be public, so there's less benefit here. But there's still some benefit, since a function taking a QuantityFactory (a pretty high level concept) should usually try to call a helper function passing `.sizer` as an argument if it needs to be accessed, since the additional things stored on `.sizer` are lower level details about indexing and a routine should avoid mixing high level things (use QuantityFactory to make them based on dim names) and low level things (manually determine shapes of arrays).

That actually makes me realize - these methods are low level routines. They shouldn't be added here.

Then I looked at the two bits of code using this. One is a routine that takes in a QuantityFactory, but then doesn't rely on the QuantityFactory to be the factory of quantities. The routine actually should be taking in a sizer, not a QuantityFactory - I'll update it to do that.

The other is the MetricTerms class, where we do have a combination of high-level quantity initialization and low-level reasoning about dimension sizes, and it makes sense to be referencing both of these.

Either way though, this is getting to be more than I bargained for in this PR - I'll revert this change, and do it in another PR.

Reply:
Updated. I still made `sizer` public, but refactoring the dycore state methods will go in another small PR.